indico/modules/rb_new/client/js/modules/bookRoom/BookRoom.jsx

```diff
@@ -275,8 +261,16 @@ class BookRoom extends React.Component {
                                     <RoomRenderer rooms={this.visibleRooms}>
                                         {room => (
                                             <Slot name="actions">
-                                                <Popup trigger={bookingModalBtn(room)}
-                                                       content={labels.bookButton}
+                                                {isAdmin && !room.isAutoConfirm && (
```

Comment:
How can I make the selector work from this page if I don't have `roomId` (obviously) available in the props of the component (`mapStateToProps`)?

Reply:
Damn that's a good question.. my first thought was to have RoomRenderer use the selector but it also doesn't have a single room... I think the problem here is that there's tons of indirection going on with RoomRenderer and the Slots.

The easiest option I see is changing the `getRoom` selector to take care of adding this information...

```js
const _getRoom = (state, {roomId}) => getAllRooms(state)[roomId];
export const getRoom = createSelector(
    _getRoom,
    isUserAdmin,
    (room, isAdmin) => {
        return {
            ...room,
            canUserBook: isUserAdmin || room.isAutoConfirm,
            canUserPrebook: !room.isAutoConfirm,
        };
    }
);
```

and later once we fetch the user's privileges from the server we'd then change the selector to something like this:

```js
const _getRoom = (state, {roomId}) => getAllRooms(state)[roomId];
export const getRoom = createSelector(
    _getRoom,
    getUserBookingPrivs,
    (room, userRoomPrivs) => {
        return {
            ...room,
            canUserBook: userRoomPrivs.canBook,
            canUserPrebook: userRoomPrivs.canPrebook,
        };
    }
);
```

Reply:
I get the idea, but I still don't get how `getRoom` can be used in this file as we don't have a single room in the props and all the checks are done here.

Reply:
It doesn't - `room` will simply contain it, since that data originally comes from `getRoom`

Reply:
As far as I can see, the `results` in BookRoom come from `getSearchResults` selector that uses `getAllRooms` but not `getRoom`.

Reply:
Indeed. Same solution though - just implement it in getAllRooms then ;)